File: mancalaje-service/src/main/java/com/jofisaes/mancala/model/Player.java
package com.jofisaes.mancala.model;

import com.jofisaes.mancala.cache.BoardManager;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.io.Serializable;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "player")
public class Player implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String name;

    @Column
    private String email;

    @Column
    private Player opponent;

    @OneToOne
    @JoinColumn(name = "id")
    private Board board;

    @Column
    private BoardManager boardManager;

    @OneToMany
    private List<Hole> allPlayerHoles;

    @OneToOne
    private Hole playerStore;
}

File: kalah-game-service/src/main/kotlin/org/jesperancinha/games/kalagameservice/model/Player.kt
package org.jesperancinha.games.kalagameservice.model

import org.hibernate.Hibernate
import org.jesperancinha.games.kalagameservice.dto.PlayerDto
import javax.persistence.*

@Entity
@Table
data class Player(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    internal val id: Long? = null,

    @Column(unique = true)
    internal val username: String? = null,

    @OneToMany(fetch = FetchType.EAGER)
    internal var boards: MutableList<Board>? = null,

    @OneToOne
    internal var opponent: Player? = null,

    @OneToOne
    internal var currentBoard: Board? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as Player

        return id != null && id == other.id
    }

    override fun hashCode(): Int = 0

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , username = $username , opponent = $opponent , currentBoard = $currentBoard )"
    }
}

val Player.toDto: PlayerDto
    get() = PlayerDto(
        id = this.id,
        username = this.username
    )
